flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/filemerging/PhysicalFile.java

```diff
@@ -88,18 +88,34 @@ PhysicalFile perform(
      */
     private boolean deleted = false;
 
+    /**
+     * If a physical file is owned by current {@link FileMergingSnapshotManager}, the current {@link
+     * FileMergingSnapshotManager} should not delete or count it if not owned.
+     */
+    private boolean isOwned;
+
     public PhysicalFile(
             @Nullable FSDataOutputStream outputStream,
             Path filePath,
             @Nullable PhysicalFileDeleter deleter,
             CheckpointedStateScope scope) {
+        this(outputStream, filePath, deleter, scope, true);
+    }
+
+    public PhysicalFile(
+            @Nullable FSDataOutputStream outputStream,
+            Path filePath,
+            @Nullable PhysicalFileDeleter deleter,
+            CheckpointedStateScope scope,
+            boolean owned) {
         this.filePath = filePath;
         this.outputStream = outputStream;
         this.closed = outputStream == null;
-        this.deleter = deleter;
+        this.deleter = owned ? deleter : null;
```

Comment:
```suggestion
        this.deleter = deleter;
```

I'd suggest keep this simple.